test/TLE.test.ts

```diff
@@ -581,6 +581,28 @@ suite("TLE", () => {
                 assert.deepStrictEqual([], pr.errors);
             });
 
+            test("with two left square brackets, literal, and closing right square bracket", () => {
+                let pr = TLE.Parser.parse("\"[[two]\"");
+                assert.notEqual(null, pr);
+                assert.equal(null, pr.leftSquareBracketToken);
+                assert.deepStrictEqual(
+                    new TLE.StringValue(TLE.Token.createQuotedString(0, "\"[[two]\"")),
+                    pr.expression);
+                assert.equal(null, pr.rightSquareBracketToken);
+                assert.deepStrictEqual([], pr.errors);
+            });
+
+            test("with literal enclosed in square brackets, string following", () => {
+                let pr = TLE.Parser.parse("\"[two]andthree\"");
+                assert.notEqual(null, pr);
+                assert.equal(null, pr.leftSquareBracketToken);
+                assert.deepStrictEqual(
+                    new TLE.StringValue(TLE.Token.createQuotedString(0, "\"[two]andthree\"")),
+                    pr.expression);
+                assert.equal(null, pr.rightSquareBracketToken);
+                assert.deepStrictEqual([], pr.errors);
+            });
+
             test("with function name without parentheses, arguments, or right square bracket", () => {
```

Comment:
Are there testcases handling an unterminated string?  If not, please add once, e.g.:

{
    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "variables": {
        "nsgId": "not terminated